### PR TITLE
Conveyor high level

### PR DESCRIPTION
**Pull request by vichik123**
We realized that there is a conveyor right before the shooter. So I decided that because it's the quickest and easiest subsystem that I would write it myself.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    ConveyorInputsAutoLogged inputs = new ConveyorInputsAutoLogged();
+
+    void setPower(double power);
+
+    void updateInputs();
+
+    @AutoLog
+    class ConveyorInputs {
+        public double setpointPower = 0;
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: `powerSetpoint`, although I'd refrain from using "setpoint" here, this term should only be used for closed-loop control contexts.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java (hunk 1 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: Well how else would I denote that this power is the wanted applied power, and not the actual?

Comment (vichik123) [reply src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: 0d292c96d376bed30857f9329910fda7f91ac569

Comment (katzuv) [reply src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: Ope- loop control is by definition "מה שיוצא אני מרוצה". There is no guarantee that the subsystem will maintain the commanded value, so what conclusions can you get from comparing the actual value to the "setpoint"?
I do think that with how easy closed-loop control is in FRC we can use it everywhere, especially in an important subsystem that feeds the shooter.

File: src/main/java/frc/robot/subsystems/conveyor/Conveyor.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static Conveyor getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(ConveyorIO io) {
+        INSTANCE = new Conveyor(io);
+    }
+
+    public Command setPower(DoubleSupplier power) {
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/conveyor/Conveyor.java]: It'd be nice to have a `feed` method which takes no parameters, I believe this'll be what we'll use most of the time.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java (hunk 1 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: > There is no guarantee that the subsystem will maintain the commanded value.

That is why we log the actual voltage as well.
I guess for the conveyor we can do PID. For literally everything else it's just overkill.

Comment (katzuv) [reply src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: > That is why we log the actual voltage as well.

And I think it's a good idea, but the setpoint value has no meaning here. If it was closed-loop controlled you could see e.g. a difference between the setpoint and the actual value and draw some conclusions ("$k_V$ should be higher"), here you won't be able to see that. Maybe you could see if your current limit is too aggressive, but that's about it.
> For literally everything else it's just overkill.

Maybe, but as I wrote somewhere already, using closed-loop control is as easy as switching a control request, running a SysId routine, and writing the values down. The tuning takes you more than 30 minutes and you didn't receive good results? Post that on the SysId Chief Delphi thread and revert to open-loop.

Comment (vichik123) [reply src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: But anywhere except the feeding it just adds some time, and doesn't help at all.

File: src/main/java/frc/robot/subsystems/conveyor/Conveyor.java (hunk 2 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/conveyor/Conveyor.java]: e7f9aaf18418aeb96fbaccafbc405b76c092fc9d

File: src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java (hunk 1 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java]: Adds some time in development or in match time? If the first, then that's why I suggest capping the time, if the latter, then it's negligible (you might even reach the setpoint faster with closed-loop control).
I think there might be a small advantage in controlling the gripper's speed to make sure the Note is held in the gripper at the same place every time. The same can be said about the intake, so the Note is transferred consistently to the gripper. However, these effects in both subsystems may be unnoticeable.

In the worst case, you're getting some software points from the judges.

**Pull request merged**